routs/soldiers.js

```diff
@@ -0,0 +1,29 @@
+const soldiers = require('../collections/soldiers.js');
+const urlib = require('url');
+
+function handleSoldierReq(req, data, done) {
+
+  const parsedUrlString = req.url.split("/");
+  if (parsedUrlString.length >= 3) {
+    done("invalid url request");
+  }
+  const urlParams = parsedUrlString[2];
+  const parsedUrl = new urlib.URL("http://localhost:3000/" + req.url);
+  if (req.method === "GET") {
+    if (parsedUrlString.length === 3 && /tt[0-9]{7}/.test(urlParams)) {
+      soldiers.findSoldier(urlParams , null, done)
+    } else if (parsedUrl.searchParams.has("name")) {
+      soldiers.findSoldier(null, parsedUrl.searchParams.get("name"), done)
+    } else if (parsedUrlString.length === 2) {
+      soldiers.findSoldier(null, null, done)
```

Comment:
It's hard to understand what and why you are sending to the `soldiers.findSoldier` function as parameters.
Find a way to pass the tests in a more informative way.
Two options I can offer you are:
1. Send variables with good names and not directly values.
2. Use "javascript's ability to send named parameters" by sending an object. e.g. soldiers.findSoldier({ param1: null, param2: null  }, done).